### DISCUSSION TITLE
Proxy Support

### DISCUSSION THREAD

**Discussion by Revibess** [ValvePython/steam#446] [General]
Has anyone managed to add proxy support to this, would be extremely helpful if you could share, i attempted to modify Wedauth.py to try and support proxies myself but I did not succeed, any ideas on how to support proxies?

**Comment by artur1214** [comment]
#### At least in WebAuth, you can just use environment variables (100% works in linux, 99% sure, works in windows too):

##### e.g.
```
$ export HTTP_PROXY="http://10.10.1.10:3128"
$ export HTTPS_PROXY="http://10.10.1.10:1080"
$ export ALL_PROXY="socks5://10.10.1.10:3434"
```

If you added this proxy variables to your system, requests lib, which used in `WebAuth`, will automatically pick up them, and will use proxy.

If you use `SteamClient`, you can also try this method, but  I almost sure, it won't work because `SteamClient` use low level python sockets under the hood. This means, for `SteamClient` you can wait untill someone rewrite it (I had this idea, but don't know how long it takes), or rewrite it by yourself :(
